src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -1,182 +1,294 @@
+    const parseToken = async () => {
+      const id = await getUserId();
+      if (id) {
+        setMyId(id);
+      }
+    };
+    parseToken();
+  }, []);
 
-            // DiscoveryAction.setNewFollowedDomains(newFollowedDomains, discoveryDispatch)
-            setFollowedDomains(newFollowedDomains)
+  const __handleOnPressDomain = (item) => {
+    const navigationParam = {
+      item: {
+        content: {
+          domain_page_id: item.domain_page_id
+        },
+        domain: {
+          image: item.logo,
+          credderScore: item?.credder_score
+        },
+        og: {
+          domain: item.domain_name,
+          domainImage: item.logo
         }
+      }
+    };
 
-        if (from === FROM_UNFOLLOWED_DOMAIN) {
-            const newUnfollowedDomains = [...unfollowedDomains]
-            newUnfollowedDomains[index].user_id_follower = willFollow ? myId : null
+    navigation.push('DomainScreen', navigationParam);
+  };
 
-            // DiscoveryAction.setNewUnfollowedDomains(newUnfollowedDomains, discoveryDispatch)
-            setUnfollowedDomains(newUnfollowedDomains)
-        }
+  const __handleFollow = async (from, willFollow, item, index) => {
+    if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
+      console.log('from 0');
+      const newFollowedDomains = [...domains];
+      newFollowedDomains[index].following = !!willFollow;
+      newFollowedDomains[index].user_id_follower = myId;
 
-        const data = {
-            domainId: item.domain_page_id,
-            source: 'discoveryScreen',
-        };
+      console.log('newFollowedDomains', newFollowedDomains[index]);
 
-        if (willFollow) {
-            await followDomain(data);
-        } else {
-            await unfollowDomain(data);
-        }
+      FollowingAction.setFollowingDomain(newFollowedDomains, followingDispatch);
+      DiscoveryAction.setDiscoveryInitialDomains(newFollowedDomains, discoveryDispatch);
     }
+    if (from === FROM_FOLLOWED_DOMAIN) {
+      console.log('from 1');
+      const newFollowedDomains = [...followedDomains];
+      newFollowedDomains[index].following = !!willFollow;
+      newFollowedDomains[index].user_id_follower = myId;
 
-    const __renderDiscoveryItem = (from, key, item, index) => <View key={`${key}-${index}`} style={styles.domainContainer}>
-        <DomainList isDomain={true}
+      setFollowedDomains(newFollowedDomains);
+    }
+
+    if (from === FROM_UNFOLLOWED_DOMAIN) {
+      console.log('from 2');
+      const newUnfollowedDomains = [...unfollowedDomains];
+      newUnfollowedDomains[index].following = !!willFollow;
+      newUnfollowedDomains[index].user_id_follower = myId;
+
+      setUnfollowedDomains(newUnfollowedDomains);
+    }
+    const data = {
+      domainId: item.domain_id_followed,
+      source: 'discoveryScreen'
+    };
+    if (willFollow) {
+      await followDomain(data);
+    } else {
+      await unfollowDomain(data);
+    }
+    if (searchText.length > 0) fetchData();
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `__handleFollow` function, there are API calls (`followDomain`, `unfollowDomain`) without error handling. If these promises reject, the error will go unhandled which could lead to unexpected behavior in your application. Always include error handling when dealing with promises:

```diff
  if (willFollow) {
-   await followDomain(data);
+   try {
+     await followDomain(data);
+   } catch (error) {
+     console.error('Error following domain:', error);
+   }
  } else {
-   await unfollowDomain(data);
+   try {
+     await unfollowDomain(data);
+   } catch (error) {
+     console.error('Error unfollowing domain:', error);
+   }
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->